recipes-core/tdx-enc-handler/tdx-enc-handler/tdx-enc.sh

```diff
@@ -97,6 +116,59 @@ tdx_enc_prepare_tpm() {
     fi
 }
 
+tdx_enc_key_recover_from_partition() {
```

Comment:
Regarding `tdx_enc_key_recover_from_partition()` and `tdx_enc_key_save_to_partition()`:

Though I'm not strongly against the approach here where the data is stored in binary format, I'm thinking about reuse of the partition for other purposes (by us or by others) and for that I'd argue that a text format would be more flexible (because we could have multiple variable size fields in any order).

A good example of the text approach would be the U-Boot environment which is kept as a series of null-terminated strings whether stored on the filesystem or on a raw device.

If you were to consider following that approach I'd suggest storing the data just like a text file with lines terminated by a newline character and the whole file being terminated by a EOT (`^D`) character. We could read the expected maximum number of blocks for the partition and then trim the data at the first occurrence of the EOT. Each line would be in the form:

```
key=value\n
```

and `value` could have anything except newlines but in general should be readable text. For example the line with the encrypted key could look like this (with the value as a hex string):

```
encpartkey=ac6722fece9d207ef8b4d3949ff2a28f0020d95624c81624e4782f405ad98072\n
```

If for whatever reason we want some kind of validation for the presence of the data in the partition we could add some fake entry at the beginning such as:

```
_MAGIC=123456\n
...entries...
```

What do you think?

Reply:
Excelent idea! I haven't though about it. Will do in V2.